cdisc_rules_engine/dataset_builders/base_dataset_builder.py

```diff
@@ -116,3 +118,14 @@ def get_define_xml_value_level_metadata(self) -> List[dict]:
     @staticmethod
     def add_row_number(dataframe: pd.DataFrame) -> None:
         dataframe["row_number"] = range(1, len(dataframe) + 1)
+
+    def get_define_metadata(self):
+        directory_path = get_directory_path(self.dataset_path)
+        define_xml_path: str = os.path.join(directory_path, DEFINE_XML_FILE_NAME)
+        define_xml_contents: bytes = self.data_service.get_define_xml_contents(
+            dataset_name=define_xml_path
+        )
+        define_xml_reader = DefineXMLReaderFactory.from_file_contents(
+            define_xml_contents, cache_service_obj=self.cache
+        )
```

Comment:
Can we move this to `get_define_xml_reader` to be shared by all of these methods?

Reply:
You mean to move to define_xml_reader_factory.py (DefineXMLReaderFactory)?  I do not see a get_define_xml_reader file. I tried to move it to DefineXMLReaderFactory, it causes 12 tests to be failed. So I am keeping it here.

Reply:
No, i mean create a method, `get_define_xml_reader` in this same file that contains this section of the current method. there are several other methods within this same file that can share the same chunk of code.

Reply:
Got it and created it.

Reply:
looks good now, except this new method can also be used by `get_define_xml_item_group_metadata` in addition to the other 2